alf/bin/verify_checkpoint.py

```diff
@@ -0,0 +1,215 @@
+import gin
+import os
+import tempfile
+import torch
+
+import alf
+from alf.trainers import policy_trainer
+from alf.algorithms.data_transformer import create_data_transformer
+from alf.environments.utils import create_environment
+from alf.trainers import policy_trainer
+from alf.utils import common, dist_utils
+import alf.utils.checkpoint_utils as ckpt_utils
+
+flags.DEFINE_string('root_dir', os.getenv('TEST_UNDECLARED_OUTPUTS_DIR'),
+                    'Root directory for writing logs/summaries/checkpoints.')
+flags.DEFINE_float('epsilon_greedy', 1., "probability of sampling action.")
+flags.DEFINE_integer('random_seed', None, "random seed")
+flags.DEFINE_integer('num_train_iterations', 2,
+                     "number of training iterations")
+flags.DEFINE_integer('num_test_steps', 10, "number of test steps")
+flags.DEFINE_string('gin_file', None, 'Path to the gin-config file.')
+flags.DEFINE_multi_string('gin_param', None, 'Gin binding parameters.')
+flags.DEFINE_string('conf', None, 'Path to the alf config file.')
+flags.DEFINE_multi_string('conf_param', None, 'Config binding parameters.')
+flags.DEFINE_float('tolerance', 0., "Allowed difference between two runs")
+
+FLAGS = flags.FLAGS
+
+
+def _step(algorithm, time_step, policy_state, trans_state, epsilon_greedy):
+    """Run one step for the algorithm."""
+    batch_size = time_step.step_type.shape[0]
+    policy_state = common.reset_state_if_necessary(
+        policy_state, algorithm.get_initial_predict_state(batch_size),
+        time_step.is_first())
+    transformed_time_step, trans_state = algorithm.transform_timestep(
+        time_step, trans_state)
+    policy_step = algorithm.predict_step(transformed_time_step, policy_state,
+                                         epsilon_greedy)
+    return policy_step, trans_state
+
+
+def _run_steps(algorithm, env, nsteps, time_steps=[]):
+    """Run several steps of algorithm.
+
+    If ``time_steps`` is provided, will use them as the input.
+    Otherwise will obtain time steps through ``env``
+    """
+    policy_state = algorithm.get_initial_predict_state(env.batch_size)
+    trans_state = algorithm.get_initial_transform_state(env.batch_size)
+    policy_steps = []
+    run_env = not bool(time_steps)
+    if run_env:
+        time_step = common.get_initial_time_step(env)
+        time_steps.append(time_step)
+    else:
+        time_step = time_steps[0]
+    for i in range(1, nsteps + 1):
+        policy_step, trans_state = _step(
+            algorithm,
+            time_step,
+            policy_state,
+            trans_state,
+            epsilon_greedy=FLAGS.epsilon_greedy)
+        if run_env:
+            time_step = env.step(policy_step.output)
+            time_steps.append(time_step)
+        else:
+            time_step = time_steps[i]
+
+        policy_steps.append(policy_step)
+        policy_state = policy_step.state
+
+    return policy_steps, time_steps
+
+
+def _create_algorithm_and_env(root_dir):
```

Comment:
should we set random seed here as well, so env randomness will also be the same?

Reply:
env randomness doesn't matter, as I store all the env timesteps so the second run will use the exact same timesteps.

Reply:
I see, this makes a more reliable test!